File: testcases/paper/Raising/Raising.cpp
#include <stdexcept>

class NetworkFailure : public std::exception {};
class DiskFailure : public std::exception {};
class InternalStatusFailure : public std::exception {};

void NetworkCheck() {
    throw NetworkFailure();
}

void DiskCheck() {
    throw DiskFailure();
}

void InternalStatusCheck() {
    throw InternalStatusFailure();
}

File: testcases/paper/Raising/Raising.py
class NetworkFailure(Exception):
    pass

class DiskFailure(Exception):
    pass

class InternalStatusFailure(Exception):
    pass

def network_check():
    raise NetworkFailure()

def disk_check():
    raise DiskFailure()

def internal_status_check():
    raise InternalStatusFailure()
